src/main/java/edu/harvard/iq/dataverse/ingest/IngestMessageBean.java

```diff
@@ -69,25 +68,30 @@ public IngestMessageBean() {
     public void onMessage(Message message) {
         IngestMessage ingestMessage = null;
 
-        Long datafile_id = null;
         AuthenticatedUser authenticatedUser = null;
         
         try {
             ObjectMessage om = (ObjectMessage) message;
             ingestMessage = (IngestMessage) om.getObject();
 
+            // if the lock was removed while an ingest was queued, ratake the lock
```

Comment:
The "if" is the first thing that addDatasetLock method does. It has some complexity and would result in the code duplication if repeated here.
If that check would be removed from the addDatasetLock method in the future without updating the code using this method, ingest code would still not break because we remove "all" ingest locks at the end (right now, there can be at most one ingest lock).
I have extended the comment to clarify this.